src/controller/CHIPDeviceController.cpp

```diff
@@ -1739,48 +1756,32 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
-                                                                                         AttributeReadFailure);
-    }
-    break;
-    case CommissioningStage::kReadSoftwareVersion: {
-        ChipLogProgress(Controller, "Reading software version");
-        BasicCluster basic;
-        SetupCluster(basic, proxy, endpoint, timeout);
-        basic.ReadAttribute<chip::app::Clusters::Basic::Attributes::SoftwareVersion::TypeInfo>(this, BasicSoftwareCallback,
-                                                                                               AttributeReadFailure);
-    }
-    break;
     case CommissioningStage::kArmFailsafe: {
-        ChipLogProgress(Controller, "Arming failsafe");
-        // TODO: should get the endpoint information from the descriptor cluster.
         GeneralCommissioning::Commands::ArmFailSafe::Type request;
-        request.expiryLengthSeconds = params.GetFailsafeTimerSeconds();
+        request.expiryLengthSeconds = params.GetFailsafeTimerSeconds().ValueOr(kDefaultFailsafeTimeout);
         request.breadcrumb          = breadcrumb;
         request.timeoutMs           = kCommandTimeoutMs;
+        ChipLogProgress(Controller, "Arming failsafe (%u seconds)", request.expiryLengthSeconds);
         SendCommand<GeneralCommissioningCluster>(proxy, request, OnArmFailSafe, OnBasicFailure, endpoint, timeout);
     }
     break;
-    case CommissioningStage::kGetNetworkTechnology: {
-        ChipLogProgress(Controller, "Sending request for network cluster feature map");
+    case CommissioningStage::kReadCommissioningInfo: {
+        ChipLogProgress(Controller, "Sending request for commissioning information");
         app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
         app::ReadPrepareParams readParams(proxy->GetSecureSession().Value());
-        app::AttributePathParams readPath(app::Clusters::NetworkCommissioning::Id,
-                                          app::Clusters::NetworkCommissioning::Attributes::FeatureMap::Id);
-        readParams.mpAttributePathParamsList    = &readPath;
-        readParams.mAttributePathParamsListSize = 1;
+
+        app::AttributePathParams readPaths[3];
+        // Read all the feature maps for all the networking clusters on any endpoint to determine what is supported
+        readPaths[0] = app::AttributePathParams(app::Clusters::NetworkCommissioning::Id,
+                                                app::Clusters::NetworkCommissioning::Attributes::FeatureMap::Id);
+        // Get the basic commissioning info from the general commissioning cluster on this endpoint (recommended failsafe time)
+        readPaths[1] = app::AttributePathParams(endpoint, app::Clusters::GeneralCommissioning::Id,
+                                                app::Clusters::GeneralCommissioning::Attributes::BasicCommissioningInfo::Id);
+        // Read all the attributes from the basic info cluster (includes vendor id / product id / software version)
```

Comment:
This could be quite a bit of data esp for products that have a ProductURL in there.  Is there a reason not to read just the three things we care about?  That would require 5 read paths here, but that seems ok....